src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -163,6 +169,12 @@ private static boolean isJoin(SQLQueryExpr sqlExpr, String sql) {
                 && ((SQLJoinTableSource) query.getFrom()).getJoinType() != SQLJoinTableSource.JoinType.COMMA;
     }
 
+    private static void performAnalysisIfEnabled(String sql) {
+        if (LocalClusterState.state().getSettingValue(QUERY_ANALYSIS_ENABLED)) {
+            new OpenDistroSqlAnalyzer().analyze(sql);
```

Comment:
Do you have to create a new analyzer for every request?

Reply:
> Do you have to create a new analyzer for every request?

Move it to field (require `static` because all methods in this class are static) may cause problem in concurrency. And also I'm thinking of pass `sql` to constructor since this class is called SQL analyzer.